### PR TITLE
Added validation for GA transfer

### PR DESCRIPTION
**Pull request by KarishmaBothara**
Validation on GA page and disable transfer if does not satisfy...
Discussed with Lillian…This needs to check two things
Balance of sender for the asset selected and 
Amount entered

Review done by Lillian

### CODE REVIEW

File: packages/app-generic-asset/src/Transfer.tsx [hunk 1]
```diff
@@ -48,6 +53,19 @@ function Transfer ({ assets, className, onClose, recipientId: propRecipientId, s
     })));
   }, [assets]);
 
+  useEffect((): void => {
+    if (assetBalance && !assetBalance.isZero()) {
```
Comment (holygits) [inline packages/app-generic-asset/src/Transfer.tsx:57]: ` setHasBalance(assetBalance && !assetBalance.isZero())`

same with the below
`setHasAvailable(amount && !amount.isZero())`

File: packages/app-generic-asset/src/Transfer.tsx [hunk 2]
```diff
@@ -54,8 +54,12 @@ function Transfer ({ assets, className, onClose, recipientId: propRecipientId, s
   }, [assets]);
 
   useEffect((): void => {
-    setHasBalance(assetBalance && !assetBalance.isZero());
```
Comment (alexsednz) [inline packages/app-generic-asset/src/Transfer.tsx]: Can you explain, why this wasn't behaving as expected?

Comment (KarishmaBothara) [reply packages/app-generic-asset/src/Transfer.tsx]: thats cause when we assetBalance && !assetBalance.isZero(), typescript didn't like as TS2345: Argument of type 'boolean | undefined' is not assignable to parameter of type 'SetStateAction<boolean>'.
cause assetBalance can be undefined.. if it is undefined, the other part of && won't be executed ever and undefined would be set as setHasBalance, when I want it to take only boolean..

Comment (alexsednz) [reply packages/app-generic-asset/src/Transfer.tsx]: You should have tried Boolean(value), have you? `setHasBalance(Boolean(assetBalance) && !assetBalance.isZero())` looks safer to me

Comment (KarishmaBothara) [reply packages/app-generic-asset/src/Transfer.tsx]: tried..it have ts error TS2532: Object is possibly 'undefined' at !assetBalance.isZero()

Comment (alexsednz) [reply packages/app-generic-asset/src/Transfer.tsx]: What about assetBalance.isDefined()?

Comment (KarishmaBothara) [reply packages/app-generic-asset/src/Transfer.tsx]: I can't add it as a part of setHasBalance.. but can add a condition if it is defined then setHasBalance.. which is what I ended up doing if (assetBalance) { ... }

Comment (alexsednz) [reply packages/app-generic-asset/src/Transfer.tsx]: The problem is the behaviour is different in (if...) unless you add the else case too

Comment (KarishmaBothara) [reply packages/app-generic-asset/src/Transfer.tsx]: changed as per our discussion..

File: packages/app-generic-asset/src/Transfer.tsx [hunk 3]
```diff
@@ -48,6 +53,11 @@ function Transfer ({ assets, className, onClose, recipientId: propRecipientId, s
     })));
   }, [assets]);
 
+  useEffect((): void => {
+    setHasBalance( (assetBalance !== undefined) && !assetBalance.isZero());
```
Comment (alexsednz) [inline packages/app-generic-asset/src/Transfer.tsx:57]: Do we format our tsx code?

**Pull request merged**